mllib/src/main/scala/org/apache/spark/ml/regression/LinearRegression.scala

```diff
@@ -327,6 +327,11 @@ class LinearRegression @Since("1.3.0") (@Since("1.3.0") override val uid: String
         throw new SparkException(msg)
       }
 
+      if (!state.actuallyConverged) {
+        logWarning("LinearRegression training fininshed but the result is not converged, " +
```

Comment:
LGTM though you could use string interpolation here.  Maybe slightly better as "Linear regression training fininshed but the result is not converged because: ..."